src/Concordium/Client/Runner.hs

```diff
@@ -2255,15 +2255,29 @@ processBakerCmd action baseCfgDir verbose backend =
       withClient backend $ do
          let senderAddr = naAddr . esdAddress . tcEncryptedSigningData $ txCfg
          AccountInfoResult{..} <- getAccountInfoOrDie senderAddr
+         energyrate <- getNrgGtuRate
+         let gtuTransactionPrice = Types.computeCost energyrate (tcEnergy txCfg) 
          case airBaker of
            Just AccountInfoBakerResult{..} -> logFatal [[i|Account is already a baker with ID #{aibiIdentity abirAccountBakerInfo}.|]]
            Nothing -> do
-             -- TODO: this should also take into account the estimated cost for this transaction
-             if airAmount < initialStake
+             if (airAmount - gtuTransactionPrice) < initialStake
              then
-               logFatal [[i|Account balance (#{showGtu airAmount}) is lower than the amount requested to be staked (#{showGtu initialStake}).|]]
+               logFatal [[i|Account balance (#{showGtu airAmount}) minus the cost of the transaction (#{showGtu gtuTransactionPrice}) is lower than the amount requested to be staked (#{showGtu initialStake}).|]]
```

Comment:
Would it be worth allowing the user to proceed anyway, if they wish to? This would mainly be useful for testing that such a transaction does fail correctly on the node.

Reply:
Added in latest commit